### PR TITLE
Fix CI (Travis -> CircleCI)

### PR DESCRIPTION
**Pull request by vinjana**
All my credits for Travis were gone and the CI didn't run anymore. CircleCI provides more credits and refills them every month (other than Travis). 

* Added mamba to retrieve conda envs for tests (more performant)
* Implemented caching of environments based on env.yml checksum.
  * This is done branch-specific, mostly to avoid race conditions. But I haven't read the full docs on this in CircleCI. Maybe that wouldn't be a problem?

A bit of pre-release preparation for 1.1.0.

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -52,6 +52,26 @@ profiles {
         conda.cacheDir = "${projectDir}/cache/conda"
     }
 
+    mamba {
```
Comment (vinjana) [inline nextflow.config:55]: This is the same as for Conda, but with `useMamba=true`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-nextflow-env-{{ .Branch }}-{{ checksum "test-environment.yml" }}
+      - restore_cache:
+          keys:
+            - v1-job-env-{{ .Branch }}-{{ checksum "task-environment.yml" }}
+      - run:
+          name: "run integration tests"
+          command: "test/test1.sh test-results/ mamba nextflowEnv/"
```
Comment (vinjana) [inline .circleci/config.yml:26]: We use mamba!

**Comment by vinjana** [comment]
@GWarsow I removed the dkfzModules profile. It's not well tested and was there merely for early development. Users should use the singularity profile.

File: nextflow.config [hunk 3]
```diff
@@ -52,6 +52,25 @@ profiles {
...
+        useMamba = true
+
+        process {
+            conda = "${projectDir}/task-environment.yml"
+
+            beforeScript = """
+            export SAMTOOLS_BINARY=samtools
+            export PICARD_BINARY=picard
+            export JAVA_BINARY=java
+            export MBUFFER_BINARY=mbuffer
```
Comment (GWarsow) [inline nextflow.config:65]: after having experienced issues with a certain mbuffer version, would it make sense to stick to a certain version here as well or is this already taken care of somewhere else?

**Review by GWarsow — APPROVED** [review]
removal of dkfzModules is fine for me

File: nextflow.config (hunk 3 above)
Comment (vinjana) [reply nextflow.config:65]: The mbuffer version is fixed in the Conda [task environment](https://github.com/DKFZ-ODCF/nf-bam2fastq/blob/6039bc2858e2168e25b1062b604b9741247a728c/task-environment.yml#L9), which is also used by the containers.

**Comment by vinjana** [comment]
Thanks, @GWarsow !

**Pull request merged**